openedx/core/lib/tests/test_request_utils.py

```diff
@@ -202,3 +210,330 @@ def test_cookie_monitoring_no_groups(self, mock_set_custom_attribute, mock_captu
+    def test_get_with_missing_reason(self, mock_logger):
+        expected_error_settings_dict = _get_expected_error_settings_dict()
+        mock_logger.error.assert_called_once_with(
+            "Skipping EXPECTED_ERRORS[%d] setting. 'REASON_EXPECTED' is required to document why %s is an expected "
+            "error.",
+            0, 'valid.module.and.class.ButMissingReason'
+        )
+        assert expected_error_settings_dict == {}
+
+    @patch('openedx.core.lib.request_utils.log')
+    @override_settings(EXPECTED_ERRORS=['not-a-dict'])
+    def test_get_with_invalid_dict(self, mock_logger):
+        expected_error_settings_dict = _get_expected_error_settings_dict()
+        mock_logger.exception.assert_called_once_with(
+            'Error processing setting EXPECTED_ERRORS. AttributeError("\'str\' object has no attribute \'get\'")'
+        )
+        assert expected_error_settings_dict == {}
+
+    @override_settings(EXPECTED_ERRORS=[{
+        'MODULE_AND_CLASS': 'test.module.TestClass',
+        'REASON_EXPECTED': 'Because'
+    }])
+    def test_get_with_defaults(self):
+        expected_error_settings_dict = _get_expected_error_settings_dict()
+        assert expected_error_settings_dict == {
+            'test.module.TestClass': {
+                'is_ignored': True,
+                'log_error': False,
+                'log_stack_trace': False,
+                'reason_expected': 'Because'
+            }
+        }
+
+
+class CustomError1(Exception):
+    pass
+
+
+class CustomError2(Exception):
+    pass
+
+
+@ddt.ddt
+class TestExpectedErrorMiddleware(unittest.TestCase):
+    """
+    Tests for ExpectedErrorMiddleware
+    """
+    def setUp(self):
+        super().setUp()
+        RequestCache.clear_all_namespaces()
+        clear_cached_expected_error_settings()
+        self.mock_request = RequestFactory().get('/test')
+        self.mock_exception = CustomError1('Test failure')
+
+    def test_get_response(self):
+        expected_response = Mock()
+
+        middleware = ExpectedErrorMiddleware(lambda _: expected_response)
+        response = middleware(self.mock_request)
+
+        assert response == expected_response
+
+    @patch('openedx.core.lib.request_utils.set_custom_attribute')
+    @patch('openedx.core.lib.request_utils.log')
+    def test_process_exception_no_expected_errors(self, mock_logger, mock_set_custom_attribute):
+        ExpectedErrorMiddleware('mock-response').process_exception(self.mock_request, self.mock_exception)
+
+        mock_logger.info.assert_not_called()
+        mock_set_custom_attribute.assert_not_called()
+
+    @patch('openedx.core.lib.request_utils.set_custom_attribute')
+    @patch('openedx.core.lib.request_utils.log')
+    @ddt.data(None, [])
+    def test_process_exception_with_empty_expected_errors(
+        self, expected_errors_setting, mock_logger, mock_set_custom_attribute,
+    ):
+        with override_settings(EXPECTED_ERRORS=expected_errors_setting):
```

Comment:
Any particular reason you're using override_settings as a context manager instead of a decorator in this test? Only asking because it doesn't match the others.

Reply:
...oh, is it because you wanted to use ddt?